diff -Naur u-boot-2a56c9f.orig/board/khadas/configs/kvim.h u-boot-2a56c9f/board/khadas/configs/kvim.h
--- u-boot-2a56c9f.orig/board/khadas/configs/kvim.h	2017-01-16 01:12:12.194421873 +0800
+++ u-boot-2a56c9f/board/khadas/configs/kvim.h	2017-01-16 01:10:04.650419427 +0800
@@ -101,7 +101,7 @@
 
 /* ddr */
 #define CONFIG_DDR_SIZE					0 //MB //0 means ddr size auto-detect
-#define CONFIG_DDR_CLK					792  //MHz, Range: 384-1200, should be multiple of 24
+#define CONFIG_DDR_CLK					768  //MHz, Range: 384-1200, should be multiple of 24
 #define CONFIG_DDR4_CLK					1100  //MHz, for boards which use different ddr chip
 /* DDR type setting
  *    CONFIG_DDR_TYPE_LPDDR3   : LPDDR3
