From ba6f3500ee9b7b984e27c73d1bf9232b4e9831a2 Mon Sep 17 00:00:00 2001
From: peak3d <pfau@peak3d.de>
Date: Mon, 10 Sep 2018 13:51:15 +0200
Subject: [PATCH] [GBM] render frames multiple times if required

---
 .../VideoRenderers/HwDecRender/RendererDRMPRIME.cpp            | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.cpp b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.cpp
index 1e850cffd8a0..5512e1d7ca18 100644
--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.cpp
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.cpp
@@ -150,9 +150,6 @@ void CRendererDRMPRIME::Update()
 
 void CRendererDRMPRIME::RenderUpdate(int index, int index2, bool clear, unsigned int flags, unsigned int alpha)
 {
-  if (m_iLastRenderBuffer == index)
-    return;
-
   CVideoBufferDRMPRIME* buffer = dynamic_cast<CVideoBufferDRMPRIME*>(m_buffers[index].videoBuffer);
   if (!buffer)
     return;
