diff --git a/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd b/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
--- /dev/null
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
@@ -175,8 +175,8 @@
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x04000000>; // 64M
-//		reserve-iomap = "true";
+                reserve-memory = <0x10000000>;
+				reserve-iomap = "true";
         };
 ///	-	picture decoder
 //$$ MODULE="picture decoder"
@@ -490,7 +490,7 @@
             gpio_dat3 = "CARD_4";
             jtag_pin = "CARD_0";
             gpio_cd = "CARD_6";
-            gpio_ro = "GPIOZ_0";
+//            gpio_ro = "GPIOZ_0";
             irq_in = <3>;
             irq_out = <5>;
             card_type = <5>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
@@ -686,56 +686,56 @@
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 2 ="reg"
 //$$ L3 PROP_STR = "pinctrl-names"
-    i2c@c1108500{ /*I2C-A*/
-		compatible = "amlogic,aml_i2c";
-		dev_name = "i2c-A";
-		status = "ok";
-		reg = <0xc1108500 0x20>;
-		device_id = <1>;
-		pinctrl-names="default";
+//    i2c@c1108500{ /*I2C-A*/
+//		compatible = "amlogic,aml_i2c";
+//		dev_name = "i2c-A";
+//		status = "ok";
+//		reg = <0xc1108500 0x20>;
+//		device_id = <1>;
+//		pinctrl-names="default";
 //		pinctrl-0=<&a_i2c_master>;
-		#address-cells = <1>;
-		#size-cells = <0>;
-		use_pio = <0>;
-		master_i2c_speed = <300000>;
-	};
+//		#address-cells = <1>;
+//		#size-cells = <0>;
+//		use_pio = <0>;
+//		master_i2c_speed = <300000>;
+//	};
 
 //$$ DEVICE="I2C_B"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 2="reg"
 //$$ L3 PROP_STR = "pinctrl-names"
 //$$ L2 PROP_CHOICE "I2C_B_pin_match" = "pinctrl-0"
-	i2c@c11087c0{ /*I2C-B*/
-		compatible = "amlogic,aml_i2c";
-		dev_name = "i2c-B";
-		status = "ok";
-		reg = <0xc11087c0 0x20>;
-		device_id = <2>;
-		pinctrl-names="default";
-		pinctrl-0=<&b_i2c_master>;
-		#address-cells = <1>;
-		#size-cells = <0>;
-		use_pio = <0>;
-		master_i2c_speed = <300000>;
-	};
+//	i2c@c11087c0{ /*I2C-B*/
+//		compatible = "amlogic,aml_i2c";
+//		dev_name = "i2c-B";
+//		status = "ok";
+//		reg = <0xc11087c0 0x20>;
+//		device_id = <2>;
+//		pinctrl-names="default";
+//		pinctrl-0=<&b_i2c_master>;
+//		#address-cells = <1>;
+//		#size-cells = <0>;
+//		use_pio = <0>;
+//		master_i2c_speed = <300000>;
+//	};
 
 //$$ DEVICE="I2C_C"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 2="reg"
 //$$ L3 PROP_STR = "pinctrl-names"
-    i2c@c11087e0{ /*I2C-C*/
-		compatible = "amlogic,aml_i2c";
-		dev_name = "i2c-C";
-		status = "ok";
-		reg = <0xc11087e0 0x20>;
-		device_id = <3>;
-		pinctrl-names="default";
+//    i2c@c11087e0{ /*I2C-C*/
+//		compatible = "amlogic,aml_i2c";
+//		dev_name = "i2c-C";
+//		status = "ok";
+//		reg = <0xc11087e0 0x20>;
+//		device_id = <3>;
+//		pinctrl-names="default";
 //		pinctrl-0=<&c_i2c_master>;
-		#address-cells = <1>;
-		#size-cells = <0>;
-		use_pio = <0>;
-		master_i2c_speed = <300000>;
-	};
+//		#address-cells = <1>;
+//		#size-cells = <0>;
+//		use_pio = <0>;
+//		master_i2c_speed = <300000>;
+//	};
 
 //$$ DEVICE="I2C_D"
 //$$ L2 PROP_STR = "status"
@@ -917,7 +917,7 @@
 			port-config = <0>; /** 0: default */
 			port-dma = <0>; /** 0: default, 1: single, 2: incr, 3: incr4, 4: incr8, 5: incr16, 6: disable*/
 			port-id-mode = <0>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
-			gpio-vbus-power = "GPIOZ_1";
+			gpio-vbus-power = "GPIODV_28";
 			gpio-work-mask	= <1>; /**0: work on pulldown,1:work on pullup*/
 			status = "okay";
 		};
@@ -1062,6 +1062,25 @@
 	};
 
 /// ***************************************************************************************
+///    -       Led
+//$$ MODULE="Led"
+//$$ DEVICE="led"
+//$$ L2 PROP_STR = "status"
+//$$ L2 PROP_U32 = "board_type"
+//$$ L2 PROP_STR = "red_led_gpio"
+//$$ L2 PROP_STR = "blue_led_gpio"
+//$$ L2 PROP_U32 = "led_mode"
+	led{
+		compatible = "amlogic-led";
+		dev_name = "aml-setio";
+		status = "okay";
+		red_led_gpio = "GPIO_TEST_N";/** if your board just need one gpio,please config red_led_gpio and blue_led_gpio the same */
+		blue_led_gpio = "GPIO_TEST_N";/**  */
+		red_led_mode = <1>; /** 1:H on L off , 0:H off L on */
+		blue_led_mode= <1>; /** 1:H on L off , 0:H off L on */
+	};
+
+/// ***************************************************************************************
 /// -   Audio
 //$$ MODULE="Audio"
 //$$ DEVICE="audio_i2s"
@@ -1798,10 +1817,10 @@
 //$$ MATCH "I2C_B_pin_match" = "&b_i2c_master"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
 //$$ L2 PROP_STR 2 = "amlogic,pins"
-		b_i2c_master:b_i2c{
-			amlogic,setmask=<5 0xc000000>;
-			amlogic,pins="GPIOZ_2","GPIOZ_3";
-		};
+//		b_i2c_master:b_i2c{
+//			amlogic,setmask=<5 0xc000000>;
+//			amlogic,pins="GPIOZ_2","GPIOZ_3";
+//		};
 
 // MATCH "I2C_C_pin_match" = "&c_i2c_master"
 // L2 PROP_U32 2 = "amlogic,setmask"
@@ -1906,7 +1925,7 @@
 /// ***************************************************************************************
 ///	-	Ethernet
         ethmac: ethernet@0xc9410000{
-                compatible = "amlogic,meson8m2-rmii-dwmac", "snps,dwmac";
+		 compatible = "amlogic,meson8m2-rgmii-dwmac", "snps,dwmac";
                 reg = <0xc9410000 0x10000
                        0xc1108108 0x4>;
                 interrupts = <0 8 1>;
