--- a/arch/arm64/boot/dts/amlogic/meson-gx-p23x-q20x.dtsi	2018-08-15 21:37:02.969149932 +0100
+++ b/arch/arm64/boot/dts/amlogic/meson-gx-p23x-q20x.dtsi	2018-08-15 22:26:03.352456178 +0100
@@ -98,7 +98,7 @@
 
 	sound {
 		compatible = "simple-audio-card";
-		simple-audio-card,name = "meson-gx-audio";
+		simple-audio-card,name = "HDMI";
 
 		assigned-clocks = <&clkc CLKID_MPLL2>,
 				  <&clkc CLKID_MPLL0>,
@@ -111,7 +111,7 @@
 		simple-audio-card,dai-link@0 {
 			/* HDMI Output */
 			format = "i2s";
-			mclk-fs = <256>;
+			mclk-fs = <128>;
 			bitclock-master = <&i2s_dai>;
 			frame-master = <&i2s_dai>;
 
@@ -175,6 +175,7 @@
 	status = "okay";
 	pinctrl-0 = <&remote_input_ao_pins>;
 	pinctrl-names = "default";
+	linux,rc-map-name = "rc-tanix-tx3mini";
 };
 
 &pwm_ef {
