--- a/arch/arm64/boot/dts/amlogic/meson-gxm-khadas-vim2.dts	2018-08-23 12:29:23.605456763 +0100
+++ b/arch/arm64/boot/dts/amlogic/meson-gxm-khadas-vim2.dts	2018-08-23 20:47:56.746587999 +0100
@@ -348,6 +348,7 @@
 &sd_emmc_a {
 	status = "okay";
 	pinctrl-0 = <&sdio_pins>;
+	pinctrl-1 = <&sdio_clk_gate_pins>;
 	pinctrl-names = "default";
 	#address-cells = <1>;
 	#size-cells = <0>;
@@ -373,6 +374,7 @@
 &sd_emmc_b {
 	status = "okay";
 	pinctrl-0 = <&sdcard_pins>;
+	pinctrl-1 = <&sdcard_clk_gate_pins>;
 	pinctrl-names = "default";
 
 	bus-width = <4>;
@@ -391,6 +393,7 @@
 &sd_emmc_c {
 	status = "okay";
 	pinctrl-0 = <&emmc_pins>, <&emmc_ds_pins>;
+	pinctrl-1 = <&emmc_clk_gate_pins>;
 	pinctrl-names = "default";
 
 	bus-width = <8>;
@@ -401,7 +404,6 @@
 	disable-wp;
 	mmc-ddr-1_8v;
 	mmc-hs200-1_8v;
-	mmc-hs400-1_8v;
 
 	mmc-pwrseq = <&emmc_pwrseq>;
 	vmmc-supply = <&vcc_3v3>;
