diff -Naur linux-7f42343.orig/init/main.c linux-7f42343/init/main.c
--- linux-7f42343.orig/init/main.c	2016-07-20 19:08:36.113415069 +0800
+++ linux-7f42343/init/main.c	2016-07-20 19:05:58.474255514 +0800
@@ -884,6 +884,8 @@
 
 static noinline void __init kernel_init_freeable(void)
 {
+	char *console = "/dev_console";
+
 	/*
 	 * Wait until kthreadd is all set-up.
 	 */
@@ -913,9 +915,12 @@
 
 	do_basic_setup();
 
-	/* Open the /dev/console on the rootfs, this should never fail */
-	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0)
-		pr_err("Warning: unable to open an initial console.\n");
+	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0) {
+		sys_mknod(console, S_IFCHR|0600, (TTYAUX_MAJOR<<8)|1);
+		if (sys_open(console, O_RDWR, 0) < 0)
+			printk(KERN_WARNING "Warning: unable to open an initial console.\n");
+		sys_unlink(console);
+	}
 
 	(void) sys_dup(0);
 	(void) sys_dup(0);
