diff -Naur linux-7f42343.orig/drivers/media/rc/meson-ir.c linux-7f42343/drivers/media/rc/meson-ir.c
--- linux-7f42343.orig/drivers/media/rc/meson-ir.c	2016-07-26 22:12:04.837956947 +0800
+++ linux-7f42343/drivers/media/rc/meson-ir.c	2016-07-26 22:12:49.637956719 +0800
@@ -82,13 +82,13 @@
 static irqreturn_t meson_ir_irq(int irqno, void *dev_id)
 {
 	struct meson_ir *ir = dev_id;
-#if !defined(CONFIG_ARCH_MESON64_ODROIDC2)
+#if 0
 	u32 duration;
 	DEFINE_IR_RAW_EVENT(rawir);
 #endif
 	spin_lock(&ir->lock);
 
-#if defined(CONFIG_ARCH_MESON64_ODROIDC2)
+#if 1
 	ir_raw_event_store_edge(ir->rc,
 		(readl(ir->reg + IR_DEC_STATUS) & STATUS_IR_DEC_IN)
 		? IR_PULSE : IR_SPACE);
