--- a/arch/arm/boot/dts/amlogic/meson8m2_n200C_2G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n200C_2G.dtd
@@ -1242,6 +1242,16 @@
 		pinctrl-0=<&remote_pins>;
 	};
 
+
+	meson-ir {
+		compatible = "amlogic,meson-gxbb-ir";
+		status = "disabled";
+		reg = <0x0 0xc8100580 0x0 0x40>;
+		interrupts = <0x0 0xc4 0x1>;
+		pinctrl-names = "default";
+		pinctrl-0 = <0x19>;
+	};
+
 /// ***************************************************************************************
 ///	-	Securitykey
 //$$ MODULE="Securitykey"
