--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
@@ -1261,6 +1261,15 @@
 		ao_baseaddr = <0xf3100480>;
 		pinctrl-names="default";
 		pinctrl-0=<&remote_pins>;
+	};
+
+	meson-ir {
+		compatible = "amlogic,meson-gxbb-ir";
+		status = "disabled";
+		reg = <0x0 0xc8100580 0x0 0x40>;
+		interrupts = <0x0 0xc4 0x1>;
+		pinctrl-names = "default";
+		pinctrl-0 = <0x19>;
 	};
 
 /// ***************************************************************************************
